docker/Dockerfile

```diff
@@ -0,0 +1,294 @@
+ARG BASE_PYPI_URL
+ARG ONEAPI_INSTALL_DIR
+ARG DPNP_BUILD_DIR=$TMPDIR/dpnp
+ARG DPNP_GIT_BRANCH
+ARG DPNP_GIT_URL
+ARG DPNP_DEBUG=1
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR
+RUN pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL $TMPDIR/dpctl*.whl \
+    && mkdir -p $DPNP_BUILD_DIR \
+    && cd $DPNP_BUILD_DIR \
+    && git clone --recursive -b $DPNP_GIT_BRANCH --depth 1 $DPNP_GIT_URL . \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    && export PATH="/opt/sycl/bin:$VIRTUAL_ENV/bin:$PATH" \
+    && export LD_LIBRARY_PATH="/opt/sycl/lib:$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH" \
+    && export DPLROOT=$ONEAPI_ROOT/dpl/latest \
+    && python setup.py clean \
+    && python setup.py build_clib \
+    && export CC=dpcpp \
+    && python setup.py build_ext \
+    # TODO: manylinux wheel build arg ?
+    && python setup.py bdist_wheel \
+    && cp dist/dpnp*.whl $TMPDIR \
+    && rm -Rf $DPNP_BUILD_DIR
+
+FROM base_builder AS numba_dpex_builder
+SHELL ["/bin/bash", "-c"]
+ARG TMPDIR
+ARG ONEAPI_INSTALL_DIR
+ARG NUMBA_DPEX_BUILD_DIR=$TMPDIR/numba_dpex
+ARG NUMBA_DPEX_GIT_BRANCH
+ARG NUMBA_DPEX_GIT_URL
+ARG INTEL_NUMBA_VERSION
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR
+RUN pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl packaging \
+    && rm -Rf $TMPDIR/* \
+    && mkdir -p $NUMBA_DPEX_BUILD_DIR \
+    && cd $NUMBA_DPEX_BUILD_DIR \
+    && git clone --recursive -b $NUMBA_DPEX_GIT_BRANCH --depth 1 $NUMBA_DPEX_GIT_URL . \
+    && apt-get update --quiet && apt-get install -y --no-install-recommends \
+        spirv-tools spirv-headers \
+    && rm -rf /var/lib/apt/lists/* \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    && export PATH=/opt/sycl/bin:$VIRTUAL_ENV/bin:$ONEAPI_INSTALL_DIR/compiler/latest/bin-llvm:$PATH \
+    && export LD_LIBRARY_PATH=/opt/sycl/lib:$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH \
+    && python setup.py bdist_wheel \
+    && cp dist/numba_dppy*.whl $TMPDIR
+
+
+# Install the 3 packages and setup the entrypoint to source the 
+# oneapi setvars.sh file on startup, and fix some other environment variables
+# accordingly.
+FROM host AS numba_dpex_dev_installer
+ARG TMPDIR
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+ARG INTEL_NUMPY_VERSION
+ARG INTEL_SCIPY_VERSION
+ARG INTEL_NUMBA_VERSION
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR
+COPY --from=numba_dpex_builder $TMPDIR/numba_dppy*.whl $TMPDIR
+RUN apt-get update --quiet && apt-get install -y --no-install-recommends spirv-tools spirv-headers \
+    && rm -Rf /var/lib/apt/lists/* \
+    && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL "numpy${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" "scipy${INTEL_SCIPY_VERSION}" $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl $TMPDIR/numba_dppy*.whl cython packaging pytest pexpect \
```

Comment:
Similarly, should we pin `cython`, `packaging`, `pytest` and `pexpect`?

Reply:
> About pinning versions for wheel, cython and scikit-build, while it doesn't cost much to expose ARGs for those (will do), I thought those packages are fairly stables and shouldn't be a concern anyway. Conda recipes on intel side do not bother with pinning those either.
Likewise for those packages